### PR TITLE
ci: wait load images inside subject before screenshot

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/demo-integrations/cypress/support/snapshot.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export interface TuiSnapshotOptions {
+    wait?: number;
+    prefixName?: string;
+    name?: string;
+    postfixName?: string;
+    capture?: 'viewport';
+}
+
+export const tuiSnapshot = (
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/support/snapshot.ts:11]: Move all these logic inside
https://github.com/Tinkoff/taiga-ui/blob/main/projects/testing/cypress/snapshot/command.ts#L5

```ts
export function tuiAddMatchImageSnapshotCommand(
    options: Options & TuiSnapshotCommandOptions,
): void {
    const matchSnapshotFn = matchImageSnapshotCommand(options);

    Cypress.Commands.add(
        'matchImageSnapshot',
        {
            prevSubject: ['optional', 'element', 'window', 'document'],
        },
        (prevSubject, name: string, options?: Options & TuiSnapshotCommandOptions) => {
            
            // any custom logic here

            matchSnapshotFn(prevSubject, name, options);
        },
    );
}
```

Comment (nsbarsukov) [reply projects/demo-integrations/cypress/support/snapshot.ts:11]: See example 
https://github.com/Tinkoff/taiga-ui/pull/1721/files#diff-224d94fd4507d13303205b318fd60c5d0a03b53dc1629ee75eb4ea4ba2fc17ebR27

Let's create command `tuiWaitAllImages` and use before `matchSnapshotFn`

File: projects/testing/cypress/snapshot/command.ts [hunk 2]
```diff
@@ -1,6 +1,65 @@
...
+
+const globalScreenshotsInfo = new Map<string, number>();
+
+function makeScreenshotName(name?: string): string {
+    const screenshot =
+        name ??
+        Cypress.currentTest.titlePath
+            .join(`-`)
+            .replace(/\s/g, `.`)
+            .replace(/'/g, ``)
```
Comment (nsbarsukov) [inline projects/testing/cypress/snapshot/command.ts:57]: What about other quotes ?

For example,
https://github.com/Tinkoff/taiga-ui/blob/d96b6d1ace5b5c3720224b489d79105161a743af/projects/demo-integrations/cypress/tests/addon-table/table-pagination/table-pagination.spec.ts#L8

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo-integrations/cypress/support/snapshot.ts (hunk 1 above)
Comment (splincode) [reply projects/demo-integrations/cypress/support/snapshot.ts]: fixed

File: projects/testing/cypress/snapshot/command.ts (hunk 2 above)
Comment (splincode) [reply projects/testing/cypress/snapshot/command.ts]: fixed

**Comment by splincode** [comment]
@vladimirpotekhin @nsbarsukov updated

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**